scripts/release.sh

```diff
@@ -87,88 +105,226 @@ while true; do
 done
 
 # Check dependencies.
-dependencies gh sort
+dependencies gh jq sort
 
 if [[ -z $increment ]]; then
 	# Default to patch versions.
 	increment="patch"
 fi
 
+# Check if the working directory is clean.
+if ! git diff --quiet --exit-code; then
+	log "Working directory is not clean, it is highly recommended to stash changes."
+	while [[ ! ${stash:-} =~ ^[YyNn]$ ]]; do
+		read -p "Stash changes? (y/n) " -n 1 -r stash
+		log
+	done
+	if [[ ${stash} =~ ^[Yy]$ ]]; then
+		maybedryrun "${dry_run}" git stash push --message "scripts/release.sh: autostash"
+	fi
+	log
+fi
+
+# Check if the main is up-to-date with the remote.
+log "Checking remote ${remote} for repo..."
+remote_url=$(git remote get-url "${remote}")
+# Allow either SSH or HTTPS URLs.
```

Comment:
Don't want to cause problems because someone did git clone git@... and another git clone https://. One might not be possible over the other depending on how environment is configured.